discord/ext/commands/core.py

```diff
@@ -2121,6 +2126,9 @@ def bot_has_permissions(**perms: bool) -> Callable[[T], T]:
     def predicate(ctx: Context) -> bool:
         guild = ctx.guild
         me = guild.me if guild is not None else ctx.bot.user
+        if ctx.channel.type == ChannelType.private:
+            return True
+
         permissions = ctx.channel.permissions_for(me)  # type: ignore
```

Comment:
Could also be done with `isinstance` on `ApplicationContext`
```suggestion
        if hasattr(ctx, 'app_permissions'):
            permissions = ctx.app_permissions
        else:
            permissions = ctx.channel.permissions_for(me)  # type: ignore
```